### PR TITLE
[Feat] `GetIntrinsic`: Add ES201x function intrinsics

### PR DESCRIPTION
**Pull request by ExE-Boss**
This populates the `%Generator*%`, `%AsyncFunction*%` and `%AsyncGenerator*%` intrinsics.

Needed by [`iterator‑helpers`](https://github.com/ExE-Boss/Iterator-Helpers).

### CODE REVIEW

File: GetIntrinsic.js [hunk 1]
```diff
@@ -7,7 +7,24 @@
 
 var undefined;
 
+var $Function = Function;
 var $TypeError = TypeError;
+var $SyntaxError = SyntaxError;
+
+var getESNextFunction = function (name, syntax) {
+	try {
+		return $Function("'use strict'; return " + syntax + ';')();
```
Comment (ljharb) [inline GetIntrinsic.js:16]: It is not really a practical option to use `Function` in this file; that ensures that nobody can ever use this library in a CSP environment (ie, it's a breaking change).

Comment (ExE-Boss) [reply GetIntrinsic.js:16]: That’s why there’s the `require('./GetIntrinsic' + name + '.js')` fallback.

Comment (ljharb) [reply GetIntrinsic.js:16]: Right, but the mere *presence* of `Function` will cause the entire app not to load in a CSP env, as far as I understand.

Comment (ExE-Boss) [reply GetIntrinsic.js:16]: That’s not how **CSP** works.

**CSP** causes `Function(...)` to throw an `EvalError`.

Comment (ljharb) [reply GetIntrinsic.js:16]: hmm, ok.

File: GetIntrinsic.js [hunk 2]
```diff
@@ -7,7 +7,24 @@
...
+	} catch (e) {
+		if (!(e instanceof $SyntaxError)) {
+			try {
+				// This will also prevent bundlers from naively including these files in the bundle:
+				// eslint-disable-next-line global-require
+				return require('./GetIntrinsic/' + name + '.js');
```
Comment (ljharb) [inline GetIntrinsic.js:22]: why use `require` at all, rather than just inlining the code into a `Function` call?

Comment (ExE-Boss) [reply GetIntrinsic.js:22]: As a fallback in case `es‑abstract` is used in an environment with [`‑‑disallow‑code‑generation‑from‑strings`](https://nodejs.org/api/cli.html#cli_disallow_code_generation_from_strings).

---

Or a bundled **CommonJS** environment that has **CSP** where the bundler was configured to allow requiring those files but load them through separate `<script>` elements.

Comment (ljharb) [reply GetIntrinsic.js:22]: I'm not concerned with node flags, lots of code breaks when those are used.

Are you saying it's a common/practical approach to automatically factor out CJS modules that use eval so they're in a separate bundle from the rest of app code?

File: test/GetIntrinsic.js [hunk 3]
```diff
@@ -5,9 +5,26 @@ var GetIntrinsic = require('../GetIntrinsic');
...
+forEach([
+	'return async function* () { var x = yield; return x || 42; }',
+	'return async function* asyncGen() { var x = yield; return x || 42; };',
+	'return { async *concise(){ var x = yield; return x || 42; } }.concise;'
+], function (fn) {
+	try {
+		// eslint-disable-next-line no-new-func
+		asyncGenFns.push(Function(fn)());
+	} catch (e) {}
+});
```
Comment (ljharb) [inline test/GetIntrinsic.js:21]:
```suggestion
var asyncGenFns = require('make-async-generator-function')();
```
:-p

**Pull request merged**